### DISCUSSION TITLE
difference between running jupyter console and basic python?

### DISCUSSION THREAD

**Discussion by IngridJSJ** [Spine-project/Spine-Toolbox#1733] [Discussion]
Hello, 

I was wondering what is difference between running a tool with jupyter console and basic python?
What is the advantage of jupyter console?

We used to run the emlabpy with jupyter console, but when adding packages for AMIRIS, then the dependencies got messed up. 
So we had to run from basic console. 
Still when running from basic console there seems to be an error : kernel died ->  we think that is because the AMIRIS dont terminate properly? This error dont come when running outside spinetoolbox
![image](https://user-images.githubusercontent.com/47012366/175300655-74ff713e-ad64-4e70-8a4a-01017d0adb34.png)
Any idea of why this kernel dies (why is there a kernel, also executing from basic console)
Luckily the process finishes and the next step is also executed, so it is fine for now. 

Similarly for running emlabpy modules (all the other tools), if I want to run with the Jupyter console, then I have to install many more packages for it to run as a jupyter console. Its not only the Ipkernel, it works when installing: 
pip install git+https://github.com/Spine-project/spinetoolbox-dev

Thank you and best regards, 
Ingrid

**Comment by IngridJSJ** [comment]
While tryin to make a kernel specification,  I get this error. 
Could you please tell me how to install the kernel specs manually? 

[23-06-2022 18:46:24] Starting Python kernel spec install process
[23-06-2022 18:46:24] 	Starting program C:/Users/isanchezjimene/.conda/envs/emlabpy/python.exe
[23-06-2022 18:46:24] 	Arguments: -m ipykernel install --user --name emlabpy --display-name emlabpy_spinetoolbox
[23-06-2022 18:46:24] 	Execution in progress...

Installing kernel specs failed. Please install them manually.
Failed


Thank you again!

**Reply by ptsavol** [reply]
Open a Conda prompt. Activate environment

`activate emlabpy`

Install ipykernel

`pip install ipykernel`

After this, in Spine Toolbox. open the Tool Specfication Editor and select the *conda-env emlabpy [Conda]* or similar as the  kernel spec for the Jupyter Console.
